commit 4b54eda2707e389b08d24e7e49399fc167837950
Author: popcornmix <popcornmix@gmail.com>
Date:   Tue Dec 6 17:05:39 2016 +0000

    bcm2835-rng: Avoid initialising if already enabled
    
    Avoids the 0x40000 cycles of warmup again if firmware has already used it

diff --git a/drivers/char/hw_random/bcm2835-rng.c b/drivers/char/hw_random/bcm2835-rng.c
index 256b0b1d0f26..cc4bcbd76b35 100644
--- a/drivers/char/hw_random/bcm2835-rng.c
+++ b/drivers/char/hw_random/bcm2835-rng.c
@@ -102,8 +102,10 @@ static int bcm2835_rng_init(struct hwrng *rng)
 	}
 
 	/* set warm-up count & enable */
-	rng_writel(priv, RNG_WARMUP_COUNT, RNG_STATUS);
-	rng_writel(priv, RNG_RBGEN, RNG_CTRL);
+	if (!(rng_readl(priv, RNG_CTRL) & RNG_RBGEN)) {
+		rng_writel(priv, RNG_WARMUP_COUNT, RNG_STATUS);
+		rng_writel(priv, RNG_RBGEN, RNG_CTRL);
+	}
 
 	return ret;
 }
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1018_4b54eda2707e389b08d24e7e49399fc167837950.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1018_4b54eda2707e389b08d24e7e49399fc167837950.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
